sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -249,4 +250,28 @@ public static String bomAwareToString(byte[] bytes, String contentType) {
             }
         }
     }
+
+    /**
+     * Prioritize {@code applicationId} set in ClientOptions over {@link HttpLogOptions}.
+     *
+     * @param logOptions provided {@link HttpLogOptions}.
+     * @param clientOptions provided {@link ClientOptions}.
+     * @return applicationId based on rule explained above.
+     *
+     */
+    public static String getApplicationId(HttpLogOptions logOptions, ClientOptions clientOptions) {
```

Comment:
I'm not fond of having this in our public API

Reply:
removed this public API.